bindings/c/foundationdb/fdb_c.h

```diff
@@ -173,6 +177,10 @@ extern "C" {
     fdb_database_create_transaction( FDBDatabase* d,
                                      FDBTransaction** out_transaction );
 
+    DLLEXPORT WARN_UNUSED_RESULT FDBFuture*
+    fdb_database_reboot_worker( FDBDatabase* db, uint8_t const* address,
+                                int address_length, fdb_bool_t check, int duration);
```

Comment:
The purpose of the `check` argument isn't clear to me. I think we might have to do some digging to understand it. @sfc-gh-etschannen do you remember why this reboot_and_check_worker thing was introduced?

Reply:
From the code, if `check=true`, then it will call `wait( checkFile->sync() );`
Not sure what this `checkFile` does, but it will wait for it to flush.

Reply:
I believe that during the time when checksums were being added to SQLite @sfc-gh-satherton wrote this to have a storage server check to see if it can find any corrupt links in the SQLite b+tree. Basically it sets a flag so that after a reboot, before starting up again a different checking function will run.

Reply:
@sfc-gh-satherton Do we need this feature in `kill`?
For the context here, do we need to expose `checkData` flag inside `RebootRequest` to the `kill` command?
The flag is used [here](https://github.com/apple/foundationdb/blob/master/fdbserver/worker.actor.cpp#L1179)
In general, do we still need it in the `RebootRequest`?

Reply:
The check-on-reboot is implemented by writing a check/validation file on disk as breadcrumb for the process to find after reboot, at which point it will eat the breadcrumb file and pass `true` to the `integrityCheck` parameter of the `openKVStore()` factory method.

I think we should preserve this capability as it is the easiest way to perform a storage engine integrity check in a live cluster without any external automation.